### DISCUSSION TITLE
Request for ability to rebuild the previous run with parameters

### DISCUSSION THREAD

**Discussion by jordan-mcgillivray005** [MCMicS/jenkins-control-plugin#635] [Ideas]
Hello, I am requesting the ability to rebuild the previous run with parameters similar to how https://plugins.jenkins.io/rebuild/ does it or possibly compatibility to interact with that plugin. We have a lot of params in our scripts so using this plugin is not as helpful for our team.

**Comment by MCMicS** [comment]
Hi
sounds great
I will have a look and think about to implement it. Reuse the plugin if it exists is one solution
another is to trigger again with last parameters manually or via Run configuration (WIP: #203)

**Comment by Guoyin-Wen** [comment]
Hi
I also have this need, and I implemented the rebuild history build through the idea scripting mechanism.
look at this: [rebuild script](https://gist.github.com/Guoyin-Wen/7d45bce77c3ac23809f78bce3abfa765)

**Reply by Guoyin-Wen** [reply]
![动画](https://github.com/user-attachments/assets/358d8b73-292a-42b6-a2c7-4f27be4ddfb4)

**Comment by MCMicS** [comment]
thanks i will have a look top integrate
